### PR TITLE
CT-1783 Submit ICO

### PR DESCRIPTION
**Pull request by stephenrichards**

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -398,14 +410,6 @@ en:
...
     response_upload_error: Errors detected with uploaded files.
     response_upload_blank?: Please select the file(s) you used in your response.
-    respond_point_of_no_return_html: |
-      <p class="notice">
-        <i class="icon icon-important">
-          <span class="visually-hidden">Important</span>
-        </i>
-        <strong>You can't update a response after marking it as sent.</strong>
-      </p>
-
```
Comment (aliuk2012) [inline config/locales/en.yml]: Why was this removed?

File: config/locales/en.yml [hunk 2]
```diff
@@ -657,14 +663,10 @@ en:
...
       heading: Request amends
     respond:
-      heading: Marking it as sent
-      reminders_html: |
-        <strong>Make sure you have:</strong>
-        <ul class="list list-bullet">
-          <li>cleared the response with the Deputy Director</li>
-          <li>uploaded the response and any supporting documents</li>
-          <li>sent the response to the person who made the request</li>
-        </ul>
```
Comment (aliuk2012) [inline config/locales/en.yml]: Why is the reminders key removed?

File: spec/features/cases/ico/respond_to_ico_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    cases_show_page.actions.mark_as_sent.click
+
+    cases_respond_page.back_link.click
+
+    expect(cases_show_page).to be_displayed(kase.id)
+
+  end
+
+end
```
Comment (aliuk2012) [inline spec/features/cases/ico/respond_to_ico_spec.rb]: Do we need this feature test if we are including it in the user journey e2e tests?

**Review by aliuk2012 — CHANGES_REQUESTED** [review]

File: config/locales/en.yml (hunk 1 above)
Comment (beccapearce) [reply config/locales/en.yml]: it wasn't in the designs and we decided to make the FOI and ICO page the same

File: config/locales/en.yml (hunk 2 above)
Comment (beccapearce) [reply config/locales/en.yml]: it wasn't in the designs and we decided to make the FOI and ICO page the same

File: app/views/cases/respond.html.slim [hunk 4]
```diff
@@ -1,8 +1,13 @@
+= GovukElementsErrorsHelper.error_summary @case.object, @case.error_summary_message, ''
+
 - content_for :page_title do
   = t('page_title.mark_as_sent_page', case_number: @case.number)
 
 - content_for :heading
   = t('.heading')
+  br
```
Comment (misaka) [inline app/views/cases/respond.html.slim]: Having these two in the heading looks a little bit odd to me:

screenshot_2018-08-03 marking response as sent - 180803001 - track-a-query

I don't see designs for it but it seems to me that other pages don't have the page heading and the case subject both so big.

**Review by misaka — CHANGES_REQUESTED** [review]

File: app/views/cases/respond.html.slim (hunk 4 above)
Comment (aliuk2012) [reply app/views/cases/respond.html.slim]: @misaka I agree with you. Its something I've been speaking to Mia about, oddly its on this page and some of the "Close cases" page.

File: config/locales/en.yml (hunk 2 above)
Comment (aliuk2012) [reply config/locales/en.yml]: 🤔  Still think this would be important for KILOs and FOI.

File: spec/features/cases/ico/respond_to_ico_spec.rb (hunk 3 above)
Comment (aliuk2012) [reply spec/features/cases/ico/respond_to_ico_spec.rb]: I think I would leave it down to the controller and model to throw and handle validation errors etc.

Comment (aliuk2012) [reply spec/features/cases/ico/respond_to_ico_spec.rb]: if we later find a bug from an unhappy path we can add it and fix it

File: app/views/cases/respond.html.slim (hunk 4 above)
Comment (beccapearce) [reply app/views/cases/respond.html.slim]: Here are the designs: https://e25x69.axshare.com/#g=1&p=ico_appeal_case_sent_to_ico
If we want to change this I suggest doing it in another PR once you've agreed a change with Mia

File: app/state_machines/configurable_state_machine/machine.rb [hunk 5]
```diff
@@ -82,7 +82,9 @@ def respond_to_missing?(method, include_private = false)
 
     def event_name(event)
       if events.include?(event.to_sym)
-        I18n.t("event.#{event}", default: event.to_s.humanize)
+        specific_key = "event.case/#{@kase.type_abbreviation.downcase}.#{event}"
+        default_key = "event.#{event}"
+        I18n.t(specific_key, default: I18n.t(default_key))
```
[the remark this answers is not in the document]
Comment (beccapearce) [reply app/state_machines/configurable_state_machine/machine.rb]: Totally agree!

File: config/locales/en.yml (hunk 2 above)
Comment (beccapearce) [reply config/locales/en.yml]: I think that this is a question for design. Once Mia is back we can discuss it with her, but I think keeping it to her designs for now is the best thing, we can always add it back in

File: app/views/cases/respond.html.slim (hunk 4 above)
Comment (misaka) [reply app/views/cases/respond.html.slim]: @beccapearce Thanks for the link. As discussed in stand-up, let's just get this through and address this separately, if at all.

**Review by misaka — APPROVED** [review]

File: config/locales/en.yml [hunk 6]
```diff
@@ -662,14 +668,10 @@ en:
...
+      heading: Mark as sent
+      respond_date: Add date this response was sent to ICO
```
Comment (aliuk2012) [inline config/locales/en.yml]: this copy is appearing for both FOI and SAR
image

File: config/locales/en.yml (hunk 2 above)
Comment (aliuk2012) [reply config/locales/en.yml]: Mia agrees that we should keep "Make sure.." copy just for FOI.

image

Ignore the action buttons in this screenshot it should have one green and one link.

Comment (beccapearce) [reply config/locales/en.yml]: I think if we're going to change the designs on this ticket we should do it as part of another ticket

Comment (aliuk2012) [reply config/locales/en.yml]: I've just added this copy back to the page and added a spec. The reason is that this change to FOI wasn't actually defined in the original ticket but it did make the ticket a little easier to implement.

**Review by aliuk2012 — APPROVED** [review]
